domain/cloudimagemetadata/modelmigration/import_test.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package modelmigration
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/juju/clock"
+	"github.com/juju/description/v8"
+	jc "github.com/juju/testing/checkers"
+	"go.uber.org/mock/gomock"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/domain/cloudimagemetadata"
+	loggertesting "github.com/juju/juju/internal/logger/testing"
+)
+
+type importSuite struct {
+	coordinator *MockCoordinator
+	service     *MockImportService
+	clock       clock.Clock
+}
+
+var _ = gc.Suite(&importSuite{})
+
+func (s *importSuite) setupMocks(c *gc.C) *gomock.Controller {
+	ctrl := gomock.NewController(c)
+
+	s.coordinator = NewMockCoordinator(ctrl)
+	s.service = NewMockImportService(ctrl)
+	s.clock = clock.WallClock
+
+	return ctrl
+}
+
+func (s *importSuite) newImportOperation(c *gc.C) *importOperation {
+	return &importOperation{
+		service: s.service,
+		logger:  loggertesting.WrapCheckLog(c),
+		clock:   s.clock,
+	}
+}
+
+// TestRegisterImport  tests the registration of import operations with the coordinator.
+func (s *importSuite) TestRegisterImport(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	s.coordinator.EXPECT().Add(gomock.Any())
+
+	RegisterImport(s.coordinator, loggertesting.WrapCheckLog(c))
+}
+
+// TestImport tests the import operation by verifying the SaveMetadata call with transformed metadata against the service.
+// It creates several different metadata into the model and check that SaveMetadata is called with the right arguments
+func (s *importSuite) TestImport(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	// Arrange
+	creationTime := time.Now()
+	defaultArgs := description.CloudImageMetadataArgs{
+		Stream:          "stream",
+		Region:          "region",
+		Version:         "version",
+		Arch:            "arch",
+		VirtType:        "virtType",
+		RootStorageType: "rootStorageType",
+		RootStorageSize: ptr(uint64(128)),
+		DateCreated:     creationTime.UnixNano(),
+		Source:          "source",
+		Priority:        40,
+		ImageId:         "attr0",
+	}
+	args := []description.CloudImageMetadataArgs{
+		suffix(defaultArgs, 0, creationTime, customSource),
+		suffix(defaultArgs, 1, creationTime, customSource),
+		suffix(defaultArgs, 2, creationTime, customSource)}
+	dst := description.NewModel(description.ModelArgs{})
+	for _, arg := range args {
+		dst.AddCloudImageMetadata(arg)
+	}
+	expectedParamsToService := transformMetadataArgsFromDescriptionToDomain(args)
```

Comment:
Same point as export about generating the expected data.

Reply:
Alright, this one is valid, generally, we shouldn't "compute" expected output from input in test...

But that's transformation code

This mostly allows to check all conditions without to much burden when reading. 

I do that to focus on the main point of the export: It's not about value transformation, but about "only export custom image". Value transformation doesn't worth crafted UT IMHO outside non-regression, which can be done through this kind of "transformStuff" code.

I may be wrong.